src/openms/include/OpenMS/DATASTRUCTURES/OSWData.h

```diff
@@ -0,0 +1,344 @@
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Chris Bielow $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/DATASTRUCTURES/String.h>
+#include <OpenMS/FORMAT/SqliteConnector.h>
+
+#include <map>
+#include <vector>
+
+namespace OpenMS
+{
+    /// high-level meta data of a transition
+    struct OPENMS_DLLAPI OSWTransition
+    {
+      public:
+        /// default c'tor
+        OSWTransition() = default;
+        /// custom c'tor which fills all the members with data; all members are read-only
+        OSWTransition(const String& annotation, const UInt32 id, const float product_mz, const char type, const bool is_decoy);
+        OSWTransition(const OSWTransition& rhs) = default;
+        OSWTransition& operator=(const OSWTransition& rhs) = default;
+        OSWTransition(OSWTransition&& rhs) = default;
+        OSWTransition& operator=(OSWTransition&& rhs) = default;
+        ~OSWTransition() = default;
+
+        /// e.g. y5/-0.002
+        const String& getAnnotation() const
+        {
+          return annotation_;
+        }
+        /// ID as used in OSWPeakGroup::transition_ids
+        UInt32 getID() const
+        {
+          return id_;
+        }
+        /// observed product m/z value
+        float getProductMZ() const
+        {
+          return product_mz_;
+        }
+        /// b, y
+        char getType() const
+        {
+          return type_;
+        }
+        /// is this a decoy transition (from a decoy protein/peptide)
+        bool isDecoy() const
+        {
+          return is_decoy_;
+        }
+
+      private:
+        String annotation_; ///< e.g. y5/-0.002
```

Comment:
alignment order of relevance?

Reply:
ordered such that padding is minimized. Because there are a lot of these...